src/github/utils/workflow-dispatch.ts

```diff
@@ -32,6 +32,17 @@ export async function dispatchWorkflow(context: GitHubContext, options: Workflow
   });
 }
 
+export async function dispatchWorker(targetUrl: string, payload: Record<string, unknown>) {
+  const result = await fetch(targetUrl, {
+    body: JSON.stringify(payload),
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
```

Comment:
I wonder if there are any special headers we should include for plugin developers. For example, org/repo information. Perhaps some type of auth mechanism etc.

Reply:
There is all the information about the event, repo, org, type, within the payload itself.

In terms of auth, I think it should be discussed. Because now we are passing down the token directly, which is mainly to avoid limits, private repo restrictions etc but is not ideal.

We could think of a granular permission system like GitHub does within actions, where plugins could request specific read / write access to certain resources, and we could subsequently pass down a token with these specific accesses in the headers. But this would be a whole new task to carry out I believe. For now plugins can authenticate themselves using their own auth, which also works fine. So I will be removing the token from the payload and instead each plugin would use their own tokens.